templates/setup.py.jj2

```diff
@@ -113,7 +113,19 @@ if python_implementation == "PyPy":
 INSTALL_REQUIRES = [
 {% for dependency in dependencies: %}
   {% if ';' not in dependency and not dependency.startswith('#'): %}
+    {% if '#egg=' in dependency: %}
+      {% set dependency = dependency.split('#egg=') %}
+      {% set repo_link, egg_name = dependency[0], dependency[1] %}
+      {% if '#' in egg_name: %}
+        {% set egg_name = egg_name.split('#')[0] %}
```

Comment:
`.strip()` repo_link and egg_name, or dependency

Reply:
`repo_link.strip()`  I think is not required, since it is always space free.

Reply:
as discussed over zulipchat, 'dependency2 #egg'.split('#')[0] will have a trailing space, and the same applies to repo_link